### PR TITLE
[crashtracker] Use blazesym to resolve stack trace names in the receiver 

### PR DESCRIPTION
**Pull request by danielsn**
# What does this PR do?

Uses blazesym to resolve symbol addresses into names.  

# Motivation

Crash reports with names are better than crashreports with pointers.

# Additional Notes

This depends on an unreleased change in Blazesym.  We pin to a commit that has the change.  Once they release it, we can move to use that version. 

# How to test the change?

run `crashtracker::api::test_crash` and see that the the emitted crash report now includes 

after:
```
  "stacktrace": [
    {
      "ip": "0xaaaab1b32350",
      "module_base_address": null,
      "names": [
        {
          "colno": 5,
          "filename": "mod.rs",
          "lineno": 66,
          "name": "backtrace::backtrace::trace_unsynchronized"
        }
      ],
      "sp": "0xffffa5f758c0",
      "symbol_address": "0xaaaab1b3231c"
    },
    {
      "ip": "0xaaaab1b32438",
      "module_base_address": null,
      "names": [
        {
          "colno": 5,
          "filename": "collectors.rs",
          "lineno": 30,
          "name": "datadog_crashtracker::collectors::emit_backtrace_by_frames"
        }
      ],
      "sp": "0xffffa5f75900",
      "symbol_address": "0xaaaab1b32380"
    },
```

Before:
```
  "stacktrace": [
    {
      "ip": "0xaaaadd44234c",
      "module_base_address": null,
      "names": null,
      "sp": "0xffffa4b508c0",
      "symbol_address": "0xaaaadd442318"
    },
    {
      "ip": "0xaaaadd442434",
      "module_base_address": null,
      "names": null,
      "sp": "0xffffa4b50900",
      "symbol_address": "0xaaaadd44237c"
    },
```

## For Reviewers
- [ ] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: crashtracker/Cargo.toml [hunk 1]
```diff
@@ -10,6 +10,10 @@ license.workspace = true
 [lib]
 crate-type = ["lib"]
 
+[target.'cfg(unix)'.dependencies]
+# TODO: move to a named release once https://github.com/libbpf/blazesym/pull/520 makes it to release
+blazesym = {git = "https://github.com/libbpf/blazesym.git", rev = "4fa15cb"}
```
Comment (r1viollet) [inline crashtracker/Cargo.toml:15]: I need to make a mental note to take the same revision

**Review by r1viollet — APPROVED** [review]
LGTM
I was also thinking about the implications.
This means that Ruby is pulling blazesym through the crashtracking functionality.
So we can merge my work on having the symbolizer available.

**Comment by r1viollet** [comment]
This is great! LGTM

**Comment by danielocfb** [comment]
FWIW, we just released `blazesym` `0.2.0-alpha.11`, which contains the changes you need, so you can use that. (for the C API you'll still need to pull from Git directly, but you could at least import a tag, if that makes it any better)

**Pull request merged**